File: analyze_reports.js
const fs = require('fs');
const path = require('path');

const reports = [
    { name: 'Admin', file: 'admin-report.json' },
    { name: 'Auth', file: 'auth-report.json' },
    { name: 'Favourites', file: 'favourites-report.json' },
    { name: 'Game', file: 'game-report.json' },
    { name: 'Profile', file: 'profile-report.json' },
    { name: 'Public', file: 'public-report.json' },
    { name: 'TalkTalk', file: 'talktalk-report.json' },
    { name: 'User', file: 'user-report.json' }
];

const basePath = 'c:\\Backup\\Desktop-20250317T120856Z-001\\Desktop\\바탕 화면\\Centennial\\SEMESTER 3\\COMP229\\BBProject\\api\\newman\\reports';

console.log('API 이름 | 총 테스트 | 성공 | 실패 | 합격률(%)');
console.log('---------|-----------|------|------|----------');

reports.forEach(report => {
    const filePath = path.join(basePath, report.file);
    const fileContent = fs.readFileSync(filePath, 'utf8');
    const json = JSON.parse(fileContent);
    const stats = json.run.stats;
    
    const total = stats.tests.total;
    const passed = stats.tests.passed;
    const failed = stats.tests.failed;
    const passRate = total > 0 ? ((passed / total) * 100).toFixed(2) : 0;
    
    console.log(`${report.name} | ${total} | ${passed} | ${failed} | ${passRate}%`);
});

File: analyze_reports.py
import json
import os

reports = [
    ('Admin', 'admin-report.json'),
    ('Auth', 'auth-report.json'),
    ('Favourites', 'favourites-report.json'),
    ('Game', 'game-report.json'),
    ('Profile', 'profile-report.json'),
    ('Public', 'public-report.json'),
    ('TalkTalk', 'talktalk-report.json'),
    ('User', 'user-report.json')
]

base_path = 'api/newman/reports'

print("API 이름 | 총 테스트 | 성공 | 실패 | 합격률(%)")
print("---------|-----------|------|------|----------")

for name, filename in reports:
    filepath = os.path.join(base_path, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        data = json.load(f)
    
    # Extract stats from run.stats
    stats = data['run']['stats']
    
    # Total tests = tests.total
    total = stats['tests']['total']
    
    # Failed tests = assertions.failed
    failed = stats['assertions']['failed']
    
    # Passed tests = total - failed
    passed = total - failed
    
    # Pass rate
    pass_rate = ((passed / total) * 100) if total > 0 else 0
    
    print(f"{name} | {total} | {passed} | {failed} | {pass_rate:.2f}%")
